test/chunked_loss/test_grpo_loss.py

```diff
@@ -0,0 +1,272 @@
+        self.grpo_loss = LigerFusedLinearGRPOFunction.apply
+        self.beta = beta
+        self.num_generations = num_generations
+
+    def forward(
+        self,
+        x,
+        attention_mask,
+        rewards,
+        ref_input=None,
+        ref_weight=None,
+        ref_bias=None,
+    ):
+        # Pass only the arguments defined in LigerFusedLinearGRPOFunction.forward()
+        return self.grpo_loss(
+            x,  # _input
+            self.lin.weight,  # weight
+            attention_mask,  # attention_mask
+            rewards,  # rewards
+            self.lin.bias,  # bias
+            ref_input,  # ref_input
+            ref_weight,  # ref_weight
+            ref_bias,  # ref_bias
+            self.beta,  # beta
+            True,  # compiled
+            ref_input is not None,  # use_ref_model
+            self.num_generations,  # num_generations
+        )
+
+
+@pytest.mark.parametrize(
+    "B, T, H, V",
+    [
+        (8, 128, 1024, 4096),
+        (3, 47, 31, 123),  # random shape
+    ],
+)
+@pytest.mark.parametrize(
+    "scalar, dtype, atol, rtol",
+    [
+        (1.0, torch.bfloat16, 5e-2, 5e-2),
+        (1.0, torch.float32, 5e-3, 5e-3),
+    ],
+)
+@pytest.mark.parametrize("bias", [True, False])
+@pytest.mark.parametrize("beta", [0.1, 0.2])
+def test_correctness(
+    B,
+    T,
+    H,
+    V,
+    scalar,
+    dtype,
+    atol,
+    rtol,
+    bias,
+    beta,
+):
+    num_generations = 4  # Fixed number of generations for testing
+    torch_lm_head_grpo = TorchLMHeadGRPO(
+        H=H,
+        V=V,
+        dtype=dtype,
+        bias=bias,
+        beta=beta,
+        num_generations=num_generations,
+    )
+    liger_lm_head_grpo = LigerLMHeadGRPO(
+        H=H,
+        V=V,
+        dtype=dtype,
+        bias=bias,
+        beta=beta,
+        num_generations=num_generations,
+    )
+
+    # Initialize weights
+    torch_lm_head_grpo.lin.weight.data = liger_lm_head_grpo.lin.weight.data = torch.randn(
+        V, H, device=device, dtype=dtype
+    )
+    if bias:
+        torch_lm_head_grpo.lin.bias.data = liger_lm_head_grpo.lin.bias.data = torch.randn(V, device=device, dtype=dtype)
+
+    # Create inputs with shape [B*num_generations, T, H]
+    _input = torch.randn(B * num_generations, T, H, device=device, dtype=dtype) * scalar
+    input1 = _input.detach().clone().requires_grad_(True)
+    input2 = _input.detach().clone().requires_grad_(True)
+
+    # Create attention mask with random padding [B*num_generations, T]
+    attention_mask = torch.ones(B * num_generations, T, device=device)
+    num_elements_to_mask = torch.randint(1, B * num_generations * T // 2, (1,)).item()
+    mask_indices = torch.randperm(B * num_generations * T)[:num_elements_to_mask]
+    attention_mask.view(-1)[mask_indices] = 0
+
+    # Create rewards with shape [B, num_generations]
+    rewards = torch.randn(B * num_generations, device=device, dtype=dtype)
```

Comment:
If it is not too much work, would it be possible to test a scenario where the rewards are same same, i.e all 1s?